discord/guild.py

```diff
@@ -895,6 +896,65 @@ def get_member(self, user_id: int, /) -> Member | None:
         """
         return self._members.get(user_id)
 
+    _FETCHABLE = TypeVar(
+        "_FETCHABLE",
+        VoiceChannel,
+        TextChannel,
+        ForumChannel,
+        StageChannel,
+        CategoryChannel,
+        Thread,
+        Member,
+    )
+
+    async def get_or_fetch(
+        self, object_type: type[_FETCHABLE], id: int, /
+    ) -> _FETCHABLE | None:
+        """Shortcut method to get data from guild object if it's cached or fetch from api if it's not.
+
+        Parameters
+        ----------
+        object_type: :class:`Type[_FETCHABLE]`
+            Type of object to fetch or get.
+
+        id: :class:`int`
+            ID of object to get.
+
+        Returns
+        -------
+
+        Optional[:class:`~Type[_FETCHABLE]`]
+            The object of type that was specified or ``None`` if not found.
+
+        Usage
+        -----
+        ``await GUILD_OBJECT.get_or_fetch(type=..., id=...)``
```

Comment:
This won't work if both parameters are positional-only.
```suggestion
        ``await GUILD_OBJECT.get_or_fetch(OBJECT_TYPE, OBJECT_ID)``
```

Reply:
Fixed usage example